lib/src/charts/widgets/pie_chart.dart

```diff
@@ -40,45 +57,40 @@ class PieChart extends StatefulWidget {
   State<PieChart> createState() => _PieChartState();
 }
 
-class _PieChartState extends State<PieChart> {
-  late RadialController _observer;
+class _PieChartState extends State<PieChart> with TickerProviderStateMixin {
+  late RadialController _controller;
 
   @override
   void initState() {
     super.initState();
-
-    var minValue = 0.0;
-    var maxValue = double.infinity;
-
-    for (var slice in widget.data.slices) {
-      minValue = min(minValue, slice.value.toDouble());
-      maxValue = max(maxValue, slice.value.toDouble());
-    }
-
-    if (minValue.isNegative) {
-      throw ArgumentError("All data values must be positive!");
-    }
-
-    // Now we can provide the chart details to the observer
-    _observer = RadialController(
-      minValue: minValue,
-      maxValue: maxValue,
+    // provide the chart details to the controller
+    _controller = RadialController(
+      targetData: [widget.data],
+      animateOnLoad: widget.animateOnLoad,
+      autoAnimate: widget.autoAnimate,
+      animation: widget.animation ??
+          AnimationController(
+            duration: widget.animDuration ?? const Duration(milliseconds: 500),
```

Comment:
Set default value in constructor of widget.
Any time you can avoid null check and assign by providing a default value in the constructor itself and making the param non nullable, always prefer that